techmatters/flex-plugins
# Customize Contacts Waiting channels

Primary reviewer: @murilovmachado 

## Description
This PR:
- Adds the option to customize the channels displayed in contacts waiting widget. This is done via a new attribute in service configuration: `contactsWaitingChannels` (typed `contactsWaitingChannels: ChannelTypes[]`, i.e. a list of channel types). If `contactsWaitingChannels` is missing in service config, all channels will be shown. If `contactsWaitingChannels` is present, it must be a list with the lowercase names of the channels the helpline wants to display (e.g. `['voice', 'web', 'twitter']`). @dee-luo @Humairaa127  [here](https://github.com/techmatters/flex-plugins/blob/ce03aeceb617da3b3bcd64dab4b23ff2918ab420/plugin-hrm-form/src/states/DomainConstants.ts#L1) you can see the valid ChannelTypes.
- Refined some types in `HrmFormPlugin.tsx`.
- Moved `QueuesStatus` to TS.
- Moved `QueueCard` to TS and Hooks API.
- Reworked `QueuesStatus` tests to use testing library.

Notes:
- In this context customize the channels means you can choose which channels to display. The internal subscription to the channels remains the same, as I don't see any benefit on disabling the subscriptions for channels that will likely not be wired to Flex anyways

### Checklist
- [ ] [Corresponding issue has been opened](https://bugs.benetech.org/browse/CHI-1085)
- [N/A] Strings are localized

### Verification steps
- `npm run dev` should show the app as usual.
- Add `contactsWaitingChannels` (can be done via `getConfig`, in `HrmFormPlugin.tsx`).
- Check that now only the channels listed in `contactsWaitingChannels` are shown.
